packages/jupyterlab-gridstack/src/editor/gridstack/widget.ts

```diff
@@ -482,17 +497,17 @@ export class GridStackWidget extends Widget {
    * Whether the dragged element is droppable or not
    *
    * @param event Event object
-   * @returns Whether the element can be dropped
+   * @returns Whether the element can be dropped or the reason why it can't
    */
-  private _isDroppable(event: IDragEvent): boolean {
+  private _isNotDroppable(event: IDragEvent): boolean | string {
```

Comment:
At first the `boolean | string` return type might look a bit odd?

Wondering whether it would be possible to use a type or interface like this for example:

```typescript
interface IDroppableTest {
   droppable: boolean;
   reason?: string;
}
```

Reply:
Thanks for the suggestion

Indeed it will make the code more obvious - I'll update the PR.

Reply:
Implemented in 7464182